gdal/ogr/ogrsf_frmts/flatgeobuf/ogr_flatgeobuf.h

```diff
@@ -144,19 +154,43 @@ class OGRFlatGeobufLayer final : public OGRLayer
         void VerifyBuffers( int bFlag ) { m_bVerifyBuffers = CPL_TO_BOOL(bFlag); }
 
         const std::string& GetFilename() const { return m_osFilename; }
+
+        static std::string GetTempFilePath(CPLString fileName, char **papszOptions);
+
+        uint16_t GetIndexNodeSize() const { return m_indexNodeSize; } 
+        OGRwkbGeometryType GetOGRGType() const { return m_eGType; } 
+};
+
+class OGRFlatGeobufEditableLayer final: public OGREditableLayer
+{
+    std::set<CPLString> m_oSetFields;
+
+    OGRFlatGeobufLayer *m_poFlatGeobufLayer;
+
+    public:
+        OGRFlatGeobufEditableLayer(OGRFlatGeobufLayer *poFlatGeobufLayer, char **papszOpenOptions);
+
+        virtual OGRErr      CreateField( OGRFieldDefn *poField,
+                                        int bApproxOK = TRUE ) override;
+        virtual OGRErr      DeleteField( int iField ) override;
+        virtual OGRErr      AlterFieldDefn( int iField, OGRFieldDefn* poNewFieldDefn, int nFlagsIn ) override;
+        virtual GIntBig     GetFeatureCount( int bForce = TRUE ) override;
+
+        const std::string& GetFilename() const { return m_poFlatGeobufLayer->GetFilename(); }
 };
 
 class OGRFlatGeobufDataset final: public GDALDataset
 {
     private:
-        std::vector<std::unique_ptr<OGRFlatGeobufLayer>> m_apoLayers;
+        std::vector<OGRLayer *> m_apoLayers;
```

Comment:
why not using a std::vector<std::unique_ptr<OGRLayer>> ? You should be able to do that with calls to .release() where needed when wrapping a OGRFlatGeobufLayer into a OGRFlatGeobufEditableLayer